game-app/game-core/src/main/java/games/strategy/engine/posted/game/pbf/NodeBbForumPoster.java

```diff
@@ -157,12 +168,18 @@ private String uploadSaveGame(
       if (status == HttpURLConnection.HTTP_OK) {
         final String json = EntityUtils.toString(response.getEntity());
         try {
-          final String url = (String) YamlReader.readMap(json).get("url");
-          return "\n[Savegame](" + url + ")";
+          final Map<String, Object> jsonObject = YamlReader.readMap(json);
+          final Map<?, ?> responseObject = (Map<?, ?>) jsonObject.get("response");
+          // This is a temporary hack to handle old versions of nodeBB forum json
+          if (responseObject == null) {
+            return (String) jsonObject.get("url");
+          }
+          final List<?> images = (List<?>) Preconditions.checkNotNull(responseObject.get("images"));
+          final Map<?, ?> imageObject = (Map<?, ?>) images.get(0);
+          return (String) Preconditions.checkNotNull(imageObject.get("url"));
```

Comment:
If the `checkNotNull` blows up, we'll get an unhelpful error message. We could help fix this by dumping the YAML contents to the error message. WDYT? Ditto for the check-not-null above (and we may want to consider doing a 'try/catch' with ClassCastException to dump the YAML again in case the format changes).

Reply:
I mean ideally the error message should never happen. Any error here would indicate the JSON output format changed again, which should hopefully not happen too often. I figured an unhelpful error message would be better than silently printing "null" as a link, but I guess you're right that including the JSON might make debugging in the future better.